core/corehttp/gateway_handler.go

```diff
@@ -38,6 +39,29 @@ type gatewayHandler struct {
 	api    coreiface.CoreAPI
 }
 
+// StatusResponseWriter enables us to override HTTP Status Code passed to
+// WriteHeader function inside of http.ServeContent.  Decision is based on
+// presence of HTTP Headers such as Location.
+type statusResponseWriter struct {
+	w http.ResponseWriter
+}
+
+func (sw *statusResponseWriter) Header() http.Header { return sw.w.Header() }
+
+func (sw *statusResponseWriter) Write(p []byte) (int, error) { return sw.w.Write(p) }
+
+func (sw *statusResponseWriter) WriteHeader(code int) {
+	// Check if we need to adjust Status Code to account for scheduled redirect
+	// This enables us to return HTTP 301
+	// for subdomain redirect in web browsers while also returning body for cli
+	// tools which does not follow redirects by default (curl, wget).
+	redirect := sw.w.Header().Get("Location")
+	if redirect != "" {
+		code = http.StatusMovedPermanently
+	}
+	sw.w.WriteHeader(code)
+}
```

Comment:
I came up with this surgical wrapper because [status code is hardcoded to `200`](https://golang.org/src/net/http/fs.go?s=4932:5036#L189) inside `http.ServeContent()`.

Is this idiomatic/acceptable in go? Not sure if we can make this change more elegant while keeping is small without duplicating `http.ServeContent`.

Reply:
Wrapping is normal. Swapping the code like this is weird but fine.